src/main/java/de/tum/in/www1/artemis/service/compass/umlmodel/parsers/ClassDiagramParser.java

```diff
@@ -0,0 +1,244 @@
+                return Optional.empty();
+            }
+
+            throw new IOException("Relationship source or target not part of model!");
+        }
+    }
+
+    /**
+     * Finds the ids of attributes in JSON representation of model and matches them to elements in map and returns the list of UMLAttribute Java objects
+     *
+     * @param classJson the json representation of model
+     * @param jsonElementMap map of element ids and elements in the model
+     * @return the list of UMLAttribute java objects
+     */
+    @NotNull
+    protected static List<UMLAttribute> parseUmlAttributes(JsonObject classJson, Map<String, JsonObject> jsonElementMap) {
+        List<UMLAttribute> umlAttributesList = new ArrayList<>();
+        for (JsonElement attributeId : classJson.getAsJsonArray(ELEMENT_ATTRIBUTES)) {
+            UMLAttribute newAttr = parseAttribute(jsonElementMap.get(attributeId.getAsString()));
+            umlAttributesList.add(newAttr);
+        }
+        return umlAttributesList;
+    }
+
+    /**
+     * Finds the ids of methods in JSON representation of model and matches them to elements in map and returns the list of UMLMethod Java objects
+     *
+     * @param objectJson the json representation of model
+     * @param jsonElementMap map of element ids and elements in the model
+     * @return the list of UMLMethod Java objects
+     */
+    @NotNull
+    protected static List<UMLMethod> parseUmlMethods(JsonObject objectJson, Map<String, JsonObject> jsonElementMap) {
+        List<UMLMethod> umlMethodList = new ArrayList<>();
+        for (JsonElement methodId : objectJson.getAsJsonArray(ELEMENT_METHODS)) {
+            UMLMethod newMethod = parseMethod(jsonElementMap.get(methodId.getAsString()));
+            umlMethodList.add(newMethod);
+        }
+        return umlMethodList;
+    }
+
+    /**
+     * Parses the given JSON representation of a UML attribute to a UMLAttribute Java object.
+     *
+     * @param attributeJson the JSON object containing the attribute
+     * @return the UMLAttribute object parsed from the JSON object
+     */
+    private static UMLAttribute parseAttribute(JsonObject attributeJson) {
+        String[] attributeNameArray = attributeJson.get(ELEMENT_NAME).getAsString().replaceAll("\\s+", "").split(":");
+        String attributeName = attributeNameArray[0];
+        String attributeType = "";
+
+        if (attributeNameArray.length == 2) {
+            attributeType = attributeNameArray[1];
+        }
+
+        return new UMLAttribute(attributeName, attributeType, attributeJson.get(ELEMENT_ID).getAsString());
+    }
+
+    /**
+     * Parses the given JSON representation of a UML method to a UMLMethod Java object.
+     *
+     * @param methodJson the JSON object containing the method
+     * @return the UMLMethod object parsed from the JSON object
+     */
+    private static UMLMethod parseMethod(JsonObject methodJson) {
```

Comment:
I think that this method is defined multiple times? Would it be possible to make it  static and define it in some utility class?